Sources/Shared/Intents/AssistIntentHandler.swift

```diff
@@ -0,0 +1,62 @@
+import Intents
+import PromiseKit
+import ObjectMapper
+
+@available(iOS 13, watchOS 6, *)
+class AssistIntentHandler: NSObject, AssistIntentHandling {
+    typealias Intent = AssistIntent
+
+    func resolveServer(for intent: Intent, with completion: @escaping (IntentServerResolutionResult) -> Void) {
+        if let server = Current.servers.server(for: intent) {
+            completion(.success(with: .init(server: server)))
+        } else {
+            completion(.needsValue())
+        }
+    }
+
+    func provideServerOptions(for intent: Intent, with completion: @escaping ([IntentServer]?, Error?) -> Void) {
+        completion(IntentServer.all, nil)
+    }
+
+    @available(iOS 14, watchOS 7, *)
+    func provideServerOptionsCollection(
+        for intent: Intent,
+        with completion: @escaping (INObjectCollection<IntentServer>?, Error?) -> Void
+    ) {
+        completion(.init(items: IntentServer.all), nil)
+    }
+
+    func handle(intent: AssistIntent, completion: @escaping (AssistIntentResponse) -> Void) {
+        guard let server = Current.servers.server(for: intent) else {
+            completion(.failure(error: "no server provided"))
+            return
+        }
+
+        struct ConversationResponse: ImmutableMappable {
+            var speech: String
+
+            init(map: Map) throws {
+                self.speech = try map.value("response.speech.plain.speech")
+            }
+        }
+
+        let promise: Promise<ConversationResponse> = Current.webhooks.sendEphemeral(
+            server: server,
+            request: .init(
+                type: "assist",
```

Comment:
Just updated the docs: https://developers.home-assistant.io/docs/api/native-app-integration/sending-data/#process-conversation

```suggestion
                type: "conversation_process",
```